$MirOS: ports/lang/llvm-gcc/patches/patch-configure_in,v 1.2 2009/12/05 13:22:48 tg Exp $
--- configure.in.orig	Thu Jan  1 00:00:00 1970
+++ configure.in	Mon Nov 10 05:26:41 2008
@@ -351,7 +351,7 @@ if test x$enable_libgomp = x ; then
     case "${target}" in
     *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu)
 	;;
-    *-*-netbsd* | *-*-freebsd* | *-*-openbsd*)
+    *-*-freebsd* | *-*-midnightbsd* | *-*-mirbsd* | *-*-netbsd* | *-*-openbsd*)
 	;;
     *-*-solaris2* | *-*-sysv4* | *-*-irix6* | *-*-osf* | *-*-hpux11*)
 	;;
@@ -426,6 +426,24 @@ case "${target}" in
     skipdirs="$skipdirs zlib fastjar target-libjava target-boehm-gc target-zlib"
     noconfigdirs="$noconfigdirs target-libgloss"
     ;;
+  *-*-midnightbsd*)
+    noconfigdirs="$noconfigdirs target-newlib target-libgloss"
+    case $target in
+      i*86-*-*) ;;
+      *)
+	noconfigdirs="$noconfigdirs ${libgcj}"
+	;;
+    esac
+    ;;
+  *-*-mirbsd*)
+    noconfigdirs="$noconfigdirs target-newlib target-libgloss"
+    case $target in
+      i*86-*-*) ;;
+      *)
+	noconfigdirs="$noconfigdirs ${libgcj}"
+	;;
+    esac
+    ;;
   *-*-netbsd*)
     # Skip some stuff on all NetBSD configurations.
     noconfigdirs="$noconfigdirs target-newlib target-libiberty target-libgloss"
@@ -1005,6 +1023,9 @@ case "${host}" in
     ;;
   *-*-sysv4*)
     host_makefile_frag="config/mh-sysv4"
+    ;;
+  i?86-*-mirbsd*)
+    # Do not use -O2 -g -fomit-frame-pointer, may be broken.
     ;;
   # This is placed last to prevent interfering with the cases above.
   i[[3456789]]86-*-*)
